ui/lobby/src/main.ts

```diff
@@ -14,10 +15,39 @@ export default function main(opts: LobbyOpts) {
   let appVNode = patch(opts.appElement, appView(ctrl));
   opts.tableElement.innerHTML = '';
   let tableVNode = patch(opts.tableElement, tableView(ctrl));
+  let counterVNode = patch(document.querySelector('.lobby__counters')!, counterView(ctrl));
 
   function redraw() {
     appVNode = patch(appVNode, appView(ctrl));
     tableVNode = patch(tableVNode, tableView(ctrl));
+    counterVNode = patch(counterVNode, counterView(ctrl));
+  }
+
+  let cols = 0;
+
+  layout(); // escape row boundary constraints in the grid without using css subgrid
+  window.addEventListener('resize', layout);
+
+  function layout() {
```

Comment:
I've tried very hard not to do layout with JS. It's a slippery slope.

It's slow. I see a massive FOUC when loading the page with slow Internet and/or device.

The page renders - badly. Then the JS kicks in and moves stuff around. Then the page has to render - again.

It's not easy to layout properly using CSS only, but it's necessary for page display performance, and so far we managed it.

Reply:
I recognize that. It is impressive that you have maintained the svelte rendering performance of craigslist in the age of electron and multi megabyte react apps.

I am terrible at spotting FOUC as I most develop from gigabit class connection, but we can fix that stuff. Connections and devices keep getting faster. I think the CSS has gotten bad enough to warrant some javascript fixup. When will it be time to let go?

Reply:
Use the browser performance tab with throttling: ![](https://i.imgur.com/MLZVqi6.png)

We're the fortunate ones with good hardware and bandwidth. Most of our players are not so lucky, and Lichess is made for them.

I'll agree to let go of CSS-only when it will have no detrimental effect to our playerbase.